src/framework/iron/api/installation/iron_repository_configuration_file.e

```diff
@@ -70,6 +72,17 @@ feature -- Access
 			end
 		end
 
+	has_repository_by_uri (a_uri: READABLE_STRING_GENERAL): BOOLEAN
+		do
+			across
+				repositories as ic
+			until
+				Result
+			loop
+				Result := ic.item.location_string.out ~ a_uri.out
```

Comment:
be careful with ~ operator ... it also compares the types.
for instance

s8: STRING_8
s32: STRING_32
imm8: IMMUTABLE_STRING_8
other: READABLE_STRING_8


s8 := "abc"
s32 := {STRING_32} "abc"
imm8 := "abc"
other := "abc"

s8 ~ s32 -> False   (STRING_8 vs STRING_32)
s8 ~ imm8-> False  (STRING_8 vs IMMUTABLE_STRING_8 ... this one is tricky)
s8 ~ other-> True   (both are STRING_8, and contains the text "abc")

Reply:
That is insane. Is Eiffel the only language where comparing strings is either impossible or superhard? How should I compare strings?

These STRINGs all the same if you would ask a random person on the street...

Reply:
I agree this is complex.
Also the fact to have STRING_8 and STRING_32 instead of just one STRING that would support unicode out of the box.

Anyway, the solution I tend to adopt is to use the  `same_string (other): BOOLEAN` function you find in the string interfaces.

But, see my last commit on iron repository, you will see how I updated your pull request.
I hope this will be helpful.

Reply:
I simply don't understand why is_equal doesn't work.

I sort of do as with Gobo's UC_STRING I ran into issues too as is_equal has the wrong postcondition I think.

But it does not make any sense unless you're extremely deeply into Eiffel.

IMO is_equal should do the right thing, problem solved.

Reply:
By the way, this is not because I commented how to do correctly with current state, that I am happy with this.
Now this discussion should be done on the forum (mailing list), rather than here.